keystore/filesystem/server_keystore.go

```diff
@@ -837,6 +841,19 @@ func (store *KeyStore) DescribeKeyFile(fileInfo os.FileInfo) (*keystore.KeyDescr
 
 	components := strings.Split(fileInfo.Name(), "_")
 
+	if len(components) == 1 {
+		id := fileInfo.Name()
+		if strings.HasSuffix(id, ".pub") {
+			id = id[:len(id)-4]
```

Comment:
I think you can `strings.TrimSuffix()` here even without checking on `.pub` existence.